airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,270 @@
+
+    """
+    # TODO: Add the description about the ZOQL EXPORT + Links
+    # TODO: Add description to class methods
+    """
+
+    logger = AirbyteLogger()
+
+    def __init__(
+        self, authenticator: Dict, url_base: str, start_date: str, window_in_days: int, client_id: str, client_secret: str, is_sandbox: bool
+    ):
+        self.authenticator = authenticator
+        self.url_base = url_base
+        self.start_date = start_date
+        self.window_in_days = window_in_days
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.is_sandbox = is_sandbox
+        self.retry_max = 3
+
+    # MAKE try/except request warper with handling errors
+    def _make_request(self, method: str = "GET", url: str = None, payload: Dict = None) -> requests.Response:
+        retry = 0
+        while retry <= self.retry_max:
+            try:
+                request = requests.request(method=method, url=url, headers=self.authenticator, json=payload)
+                request.raise_for_status()
+                return request
+            except requests.exceptions.HTTPError as e:
+                retry += 1
+                if retry <= self.retry_max:
+                    self._handle_requests_exceptions(e)
+                    continue
+                else:
+                    raise Exception(e)
+
+    # MAKE QUERY: for data_query_job
+    def _make_dq_query(
+        self, q_type: str = "select", obj: str = None, date_field: str = None, start_date: str = None, end_date: str = None
+    ) -> Dict:
+        valid_types = ["select", "describe", "show_tables"]
+        base_query = {"compression": "NONE", "output": {"target": "S3"}, "outputFormat": "JSON"}
+
+        if q_type not in valid_types:
+            raise ZOQLQueryNotValid(q_type, valid_types)
+        elif q_type == "select":
+            end_date = pendulum.now().to_datetime_string() if end_date is None else end_date
+            base_query[
+                "query"
+            ] = f"select * from {obj} where {date_field} >= TIMESTAMP '{start_date}' and {date_field} <= TIMESTAMP '{end_date}'"
+        elif q_type == "describe":
+            base_query["query"] = f"DESCRIBE {obj}"
+        elif q_type == "show_tables":
+            base_query["query"] = "SHOW TABLES"
+        return base_query
+
+    # SUBMIT: data_query_job using data_query
+    def _submit_dq_job(self, dq_query: Dict) -> str:
+        submit_job = self._make_request(method="POST", url=f"{self.url_base}/query/jobs", payload=dq_query)
+        # self.logger.debug(submit_job.json()["data"]["id"])
+        return submit_job.json()["data"]["id"]
+
+    # CHECK: the submited data_query_job status
+    def _check_dq_job_status(self, dq_job_id: str) -> Dict:
+        # Define the job error statuses
+        errors = ["failed", "canceled", "aborted"]
+        # Error msg: the cursor_field cannot be resolved
+        cursor_error = "Column 'updateddate' cannot be resolved"
```

Comment:
While we reading the stream, we try to read it using the default `cursor_field`, which is `updateddate`.
Since the read method doesn't know which stream has the default cursor_field and which hasn't - we need to switch that dynamically if it's not present in the stream's columns. So, yes, we still have this exception, and we need that to switch the cursor_field correctly.

Reply:
cursor_field  is identified when we get schema